### PR TITLE
ContainerLogsV2 and Basic Logs support

### PR DESCRIPTION
**Pull request by samaea**
## PR Summary

Addresses issue https://github.com/Azure/AKS-Construction/issues/437. 

In addition, added:-

- updateDependencyFn() function (under addonsTabs) that assists with enforcing dependency between two features (e.g. In order to activate Basic Logs on Container Logs V2, you also need to enable the Container Logs V2 schema. It might be worth considering to make this a global function instead (i.e same place as where updateFn() is defined).
- Under deployTabs, preview_post_params array has been added to support flagging preview features passed as params to the postdeploy script. At present, we are only capturing preview params under ARM/Bicep.

image

image

## PR Checklist

- [X] PR has a meaningful title
- [X] Summarized changes
- [X] This PR is ready to merge and is not **Work in Progress**
- [X] Link to a filed issue
- [X] Screenshot of UI changes (if PR includes UI changes)

### CODE REVIEW

File: bicep/main.bicep [hunk 1]
```diff
@@ -1514,6 +1517,17 @@ resource aks_law 'Microsoft.OperationalInsights/workspaces@2022-10-01' = if (cre
   )
 }
 
+
+resource containerLogsV2_Basiclogs 'Microsoft.OperationalInsights/workspaces/tables@2022-10-01' = if(containerLogsV2BasicLogs){
+  name: '${aks_law_name}/ContainerLogV2'
+  properties: {
+    plan: 'Basic'
+  }
+  dependsOn: [
```
Comment (Gordonby) [inline bicep/main.bicep:1526]: Do we need the log analytics table creation to wait for the AKS cluster?
I'd hope it's not connected, and we can drop the dependsOn.

File: postdeploy/scripts/certmanager-install.sh [hunk 2]
```diff
@@ -1,11 +0,0 @@
-#!/bin/bash
```
Comment (Gordonby) [inline postdeploy/scripts/certmanager-install.sh:1]: why has this file been removed?

Comment (samaea) [reply postdeploy/scripts/certmanager-install.sh]: Great catch! Readded certmanager-install.sh as per offline discussion.

**Comment by samaea** [comment]
👋 @khowling, incorporated feedback from our PR discussion. Also added another function in addons.js to ensure any features dependent on the ContainerLogsV2 schema are disabled as well (at present it only accommodates for the Basic Logs feature). This avoids the scenario where the user selects on Basic Logs (which automatically selects the ContainerLogsV2 feature) and then decides to disable the ContainerLogsV2 feature (i.e safety guard in both directions now).

**Comment by khowling** [comment]
hey @samaea , looking good!  I just tested the UI, and one small thing,  in Monitoring, If I select 'Azure Monitor for Containers', then select the V2 Logs, the parameters appear on the deployment command as expected. But it I then select 'None' for Monitoring, the V2 log parameters still appear.  To fix this, can you add an additional check in the deployment logic, take a look at this as an example 
image

**Comment by samaea** [comment]
> hey @samaea , looking good! I just tested the UI, and one small thing, in Monitoring, If I select 'Azure Monitor for Containers', then select the V2 Logs, the parameters appear on the deployment command as expected. But it I then select 'None' for Monitoring, the V2 log parameters still appear. To fix this, can you add an additional check in the deployment logic, take a look at this as an example image

Hey @khowling - great catch! All done. :)

**Review by khowling — APPROVED** [review]

**Pull request merged**